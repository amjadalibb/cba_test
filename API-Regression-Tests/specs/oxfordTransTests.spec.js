const { requestAPI } = require('../support/api.js');
const { expect } = require('chai');
const { expectSnapshot, httpStatus, report } = require('../support');
const { SCENARIOS } = require('../tests/oxfordTransTests.js')

describe('Oxford Dictionary Tests for Translations', function() {
	it('Get translation for word test', async function() {
		report(this, 'API Options BaseURL', SCENARIOS.optionsValid.baseURL);
		report(this, 'API Options Endpoint', SCENARIOS.optionsValid.endpoint);
		const response = await requestAPI(SCENARIOS.optionsValid);
		report(this, 'API response', response.data.results[0].lexicalEntries[0].entries[0].senses[0].subsenses[0].translations[0].text);
		expect(response.status).to.equal(httpStatus.OK);
		expectSnapshot(response.data.results[0].lexicalEntries[0].entries[0].senses[0].subsenses[0].translations[0].text).toMatchSnapshot(this);
	});

	it('Get translation for an invalid word ThisWordDoesNotExist', async function() {
		report(this, 'API Options BaseURL', SCENARIOS.optionsInvalid.baseURL);
		report(this, 'API Options Endpoint', SCENARIOS.optionsInvalid.endpoint);
		const response = await requestAPI(SCENARIOS.optionsInvalid);
		report(this, 'API response', response.data);
		expect(response.status).to.equal(httpStatus.NOT_FOUND);
		expectSnapshot(response.data).toMatchSnapshot(this);
	});

	it('Bad request on getting entries of word', async function() {
		report(this, 'API Options BaseURL', SCENARIOS.optionsBadRequest.baseURL);
		report(this, 'API Options Endpoint', SCENARIOS.optionsBadRequest.endpoint);
		report(this, 'API Options Params', SCENARIOS.optionsBadRequest.params);
		const response = await requestAPI(SCENARIOS.optionsBadRequest);
		report(this, 'API response', response.data);
		expect(response.status).to.equal(httpStatus.BAD_REQUEST);
		expectSnapshot(response.data).toMatchSnapshot(this);
	});
});