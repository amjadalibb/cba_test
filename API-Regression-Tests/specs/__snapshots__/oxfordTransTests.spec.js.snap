// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Oxford Dictionary Tests for Translations Bad request on getting entries of word 1`] = `
Object {
  "error": "Unknown filter value 'UnknownFilter' Please, consult valid values in the utility endpoints.",
}
`;

exports[`Oxford Dictionary Tests for Translations Get translation for an invalid word ThisWordDoesNotExist 1`] = `
Object {
  "error": "No entry found matching supplied source_lang target_lang and word",
}
`;

exports[`Oxford Dictionary Tests for Translations Get translation for word test 1`] = `"prueba"`;
